src/core/transpile.jl

```diff
@@ -914,6 +914,41 @@ function cast_to_rz_rx_rz(gate::Universal, target::Int)::Vector{Gate}
     return gate_array
 end
 
+# Decompose a Universal gate as U = exp(im*α)AXBXC.
+# See: Nielsen and Chuang, Quantum Computation and Quantum Information, p175.
+function decompose_universal_to_A_B_C_gates(
+    gate::Universal,
+    target::Int,
+)::Tuple{Real,Gate,Gate,Gate}
+    params = get_gate_parameters(gate)
+
+    γ = params["theta"]
+    β = params["phi"]
+    δ = params["lambda"]
+
+    A = rotation_z(β) * rotation_y(γ / 2.0)
+
+    B = rotation_y(-γ / 2.0) * rotation_z(-(δ + β) / 2.0)
+
+    # numerical stability issues arise for rotation_z with vanishingly small phase angle
+    if abs(δ - β) > sqrt(eps(typeof(δ)))
```

Comment:
I don't undestand how this gives numerical instability issues.

Given an approximate Taylor series expansion about zero of `exp(-im * x)` (The exponent of the `rotation_z`)

![image](https://github.com/SnowflurrySDK/Snowflurry.jl/assets/99685336/a73ffe11-8863-4db8-8950-5f1ae5433ee6)

I get identical results

![image](https://github.com/SnowflurrySDK/Snowflurry.jl/assets/99685336/3f643683-bd3b-48b9-abd0-685369329ca7)

Could you add a comment on why this gives numerical instability issues, as it's not clear from these 5 lines to me?

Reply:
I took it out, and the error i was seeing is gone